collection-scripts/gather_db

```diff
@@ -28,26 +48,35 @@ if [ "${OPENSTACK_DATABASES-unset}" = "unset" ] || [[ -z "${OPENSTACK_DATABASES}
     return
 fi
 
+if [ "${FILTER_DB_CELLS-unset}" = "unset" ] || [[ -z "${FILTER_DB_CELLS}" ]]; then
+    # do not skip cells db if not explicitly set
+    FILTER_DB_CELLS="false"
+fi
+
 # Create the db_dump directory in the BASE_COLLECTION_PATH
 mkdir -p "$DB_DUMP"
 data=$(/usr/bin/oc get openstackcontrolplane --all-namespaces -o go-template='{{range $ins,$service := .items}}{{printf "%s %s\n" $service.metadata.namespace $service.spec.secret}}{{end}}')
 while read -r namespace secret; do
     [[ -z "$namespace" || -z "$secret" ]] && break
-    # get the pwd used to run mysqldump from the galera pod
+    mkdir -p "$DB_DUMP/$namespace"
+    # get the pwd used to run mysqldump from the galera pod in the current namespace
     dbpass=$(/usr/bin/oc get -n $namespace secret/"$secret" -o go-template='{{ index .data "AdminPassword" | base64decode }}')
-    # select the (first) galera pod (and exclude the galera-cellX database services)
-    dbpod="$(/usr/bin/oc -n $namespace get pod -l app=galera --no-headers -o=custom-columns=NAME:.metadata.name | grep -v cell | tr '\n' ' ' | awk '{print $1}')"
-    if [[ "$OPENSTACK_DATABASES" == "ALL" ]]; then
-        DB_OPT="$DB_OPT --all-databases"
-        # dump all databases
-        dump_db "$dbpod" "$namespace" "$dbpass"
-    else
-        # Convert the database list to an array
-        IFS=',' read -r -a OPENSTACK_DATABASES <<< "$OPENSTACK_DATABASES"
-        # for each service dump the associated database (if exists)
-        for service in "${OPENSTACK_DATABASES[@]}"; do
-            dump_db "$dbpod" "$namespace" "$dbpass" "$service"
-        done
-    fi
+    # get the list of galera pods in the current namespace
+    dbpods="$(get_dbpod $namespace)"
+    for dbinst in $dbpods; do
+        if [[ "$OPENSTACK_DATABASES" == "ALL" ]]; then
```

Comment:
This block is unchanged, it selects the first galera Pod for each statefulSet on the same namespace, and dumps the databases according to `OPENSTACK_DATABASES`.